src/python/tests/storage/test_storage_api.py

```diff
@@ -0,0 +1,265 @@
+        assert res["rows"][0]["value"] == 10
+        assert res["rows"][0]["previous_value"] == 2
+
+    @pytest.mark.skip(reason="FOGL-615")
+    def test_where_invalid_key(self):
+        res = Storage().query_tbl_with_payload("statistics", PayloadBuilder().WHERE(["bla", "=", "invalid"]).payload())
+        assert "ERROR" in res["message"]
+
+    @pytest.mark.skip(reason="Payload builder does not parse more than 1 AND_WHERE correctly")
+    def test_multiple_and_where(self):
+        res = Storage().query_tbl_with_payload("statistics", PayloadBuilder().
+                                               WHERE(["value", "!=", 0]).
+                                               AND_WHERE(["key", "=", "TEST_1"]).
+                                               AND_WHERE(["previous_value", "<", 10]).payload())
+        assert len(res["rows"]) == 1
+        assert res["count"] == 1
+        assert res["rows"][0]["key"] == "TEST_1"
+        assert res["rows"][0]["description"] == "Testing the storage service data 1"
+        assert res["rows"][0]["value"] == 10
+        assert res["rows"][0]["previous_value"] == 2
+
+    @pytest.mark.skip(reason="Payload builder does not parse more than 1 OR_WHERE correctly")
+    def test_multiple_or_where(self):
+        res = Storage().query_tbl_with_payload("statistics", PayloadBuilder().
+                                               WHERE(["value", "=", 0]).
+                                               OR_WHERE(["key", "=", "TEST_1"]).
+                                               OR_WHERE(["previous_value", ">", 10]).payload())
+        assert len(res["rows"]) == 1
+        assert res["count"] == 1
+        assert res["rows"][0]["key"] == "TEST_1"
+        assert res["rows"][0]["description"] == "Testing the storage service data 1"
+        assert res["rows"][0]["value"] == 10
+        assert res["rows"][0]["previous_value"] == 2
+
+    def test_limit(self):
+        res = Storage().query_tbl_with_payload("statistics", PayloadBuilder().LIMIT(1).payload())
+        assert len(res["rows"]) == 1
+        assert res["count"] == 1
+        assert res["rows"][0]["key"] == "TEST_1"
+        assert res["rows"][0]["description"] == "Testing the storage service data 1"
+        assert res["rows"][0]["value"] == 10
+        assert res["rows"][0]["previous_value"] == 2
+
+    @pytest.mark.skip(reason="Payload builder does not support offset without limit")
+    def test_offset(self):
+        pass
+
+    def test_limit_offset(self):
+        res = Storage().query_tbl_with_payload("statistics", PayloadBuilder().LIMIT(2).OFFSET(1).payload())
+        assert len(res["rows"]) == 1
+        assert res["count"] == 1
+        assert res["rows"][0]["key"] == "TEST_2"
+        assert res["rows"][0]["description"] == "Testing the storage service data 2"
+        assert res["rows"][0]["value"] == 15
+        assert res["rows"][0]["previous_value"] == 2
+
```

Comment:
Should we add test_skip and test_limit_skip?

Reply:
Both skip and offset generates the same JSON payload. Purpose of these tests was to see the integration with storage layer. Since we have already tested that skip and offset generates the same json in test_payload_builder.py there was no point to test the same payload with storage layer again. That is why skipped